be/src/exprs/agg/map_agg.h

```diff
@@ -0,0 +1,151 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <fmt/format.h>
+
+#include "column/binary_column.h"
+#include "column/column.h"
+#include "column/column_helper.h"
+#include "column/fixed_length_column.h"
+#include "column/hash_set.h"
+#include "column/map_column.h"
+#include "column/type_traits.h"
+#include "exprs/agg/aggregate.h"
+#include "exprs/function_context.h"
+#include "gutil/casts.h"
+#include "util/phmap/phmap.h"
+#include "util/time.h"
+
+namespace starrocks {
+
+template <LogicalType KT, typename MyHashMap = std::map<int, Datum>>
+struct MapAggAggregateFunctionState : public AggregateFunctionEmptyState {
+    using KeyColumnType = RunTimeColumnType<KT>;
+    using KeyType = typename SliceHashSet::key_type;
+
+    MyHashMap hash_map;
+    void update(MemPool* mem_pool,
```

Comment:
It seems other system implement has a max limit for map_agg

Reply:
I'm not sure. It seems that the map_agg of Trino does not have a limit, all my implement is simulating the one in Trino.